### ISSUE TITLE
Negative values provided to the `Get-Content` `-Tail` parameter no longer return the entire content, and instead print an error

### ISSUE THREAD

**Issue by AlexFolland** [PowerShell/PowerShell#21347]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Negative values for `Get-Content` parameter `-Tail` are no longer considered valid, despite the documentation specifying that the accepted type is Int32 which supports negative values, and that providing a negative value results in an unlimited number of lines, effectively showing the full content of the file regardless of its length, even if that length is above `$([int]::MaxValue)`.

The [latest documentation for this command](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.management/get-content?view=powershell-7.5) says this in the `-Tail` section:

> Negative values cause the cmdlet to return the entire contents.

So, it looks like the current behavior does not match the documented behavior.

I see no current equivalent way to guarantee getting a truly unlimited number of lines beyond `$([int]::MaxValue)`.  That returns a very large number, but does not guarantee all lines if the file has more lines than `$([int]::MaxValue)`.

### Expected behavior

```console
PS> Get-Content ${filePath} -Wait -Tail -1
<full content of the file at ${filePath} shown here>
```
```


### Actual behavior

```console
PS> Get-Content ${filePath} -Wait -Tail -1
Get-Content: Cannot validate argument on parameter 'Tail'. The -1 argument is less than the minimum allowed range of 0. Supply an argument that is greater than or equal to 0 and then try the command again.
```
```


### Error details

```console
I am submitting this issue report on a different computer from the one in which I discovered this issue, so I don't have direct copy access to the output of `Get-Error`.  The error says `Get-Content: Cannot validate argument on parameter 'Tail'. The -1 argument is less than the minimum allowed range of 0. Supply an argument that is greater than or equal to 0 and then try the command again.` though.
```

### Environment data

```powershell
I am submitting this issue report on a different computer from the one in which I discovered this issue, so I don't have direct copy access to the output of ${PSVersionTable}.  It says the version is `7.4.1` though.
```

**Comment by rhubarb-geek-nz** [comment]
>  That returns a very large number, but does not guarantee all lines if the file has more lines than $([int]::MaxValue).

Just to put in perspective

PS> [int]::MaxValue
2147483647

so that is 2,147,483,647

There are only 783,137 _words_ in the King James bible and the Linux kernel is of the order of 30 million lines across all files.

**Comment by AlexFolland** [comment]
Still, if almost every character is a line break, that's only a 2 GiB file.  I have dealt with logs in the hundreds of gigabytes before, and introducing an arbitrary limitation when there previously wasn't one is a regression, regardless of perspective.  Also, the documentation not matching the behavior is itself an issue.

**Comment by rhubarb-geek-nz** [comment]
If you want all the content of the file, then why use the Tail argument?

The danger of the Tail argument with a very large number is if the cmdlet has to store those billions of _lines_ in memory before determining that it finally has the last one and can start writing them to the output.

**Comment by AlexFolland** [comment]
The fact is that there's a regression and the behavior doesn't match the documentation.  That's the purpose of the report.  Please stop commenting on this issue if the comments are not intended to further the issue's resolution.

**Comment by MartinGC94** [comment]
The change was made here: https://github.com/PowerShell/PowerShell/pull/19715 looking at the old code it seems that there was never any reason to specify a negative number because if you did then it would just read it from start to finish, as if you hadn't specified `Tail` at all. So for the statement/question:

> I see no current equivalent way to guarantee getting a truly unlimited number of lines

The answer seems to be to just not specify `Tail` at all.

**Comment by AlexFolland** [comment]
OK, thanks.  I've tested that, and it does seem to work as desired.  I'll update my command to omit `-Tail` and its value.  My particular use case is solved, but the issue with the documentation remains.

It sounds like the resolution for this issue, then, is to update the documentation to remove the reference to using negative value, and possibly suggest omitting `-Tail` to accomplish the desired behavior.

**Comment by rhubarb-geek-nz** [comment]
> Edit: There was a comment above this after my previous comment, but it seems to have been deleted.

Yes, the comment that offended you suggested that a tail of infinite length made no sense and it would be better to not use the tail argument at all if that was the effect you wanted.

**Comment by mklement0** [comment]
I've created a docs issue:

* https://github.com/MicrosoftDocs/PowerShell-Docs/issues/10957

**Comment by AlexFolland** [comment]
The documentation change does not suggest omitting `-Tail` to accomplish the desired behavior to bridge the gap for users who previously passed a negative value to `-Tail`, but those users can find this ticket if they need to.  Either way, this issue is now technically resolved, so I'm closing it.  Thanks for the effort!

**Comment by AlexFolland** [comment]
> 📣 Hey @AlexFolland, how did we do? We would love to hear your feedback with the link below! 🗣️
> 
> 🔗 https://aka.ms/PSRepoFeedback
> 
> > **[Microsoft Forms](https://aka.ms/PSRepoFeedback)**

I already gave my feedback in the ticket.  If someone cares about it, they can read the ticket.
